common/djangoapps/student/views.py

```diff
@@ -699,12 +701,20 @@ def dashboard(request):
     course_enrollments = list(get_course_enrollments(user, site_org_whitelist, site_org_blacklist))
 
     # Get the entitlements for the user and a mapping to all available sessions for that entitlement
+    # If an entitlement has no available sessions, pass through a mock course overview object
     course_entitlements = list(CourseEntitlement.get_active_entitlements_for_user(user))
     course_entitlement_available_sessions = {}
+    unfulfilled_entitlement_mock_sessions = {}
     for course_entitlement in course_entitlements:
         course_entitlement.update_expired_at()
-        valid_course_runs = get_visible_course_runs_for_entitlement(course_entitlement)
-        course_entitlement_available_sessions[str(course_entitlement.uuid)] = valid_course_runs
+        available_sessions = get_visible_sessions_for_entitlement(course_entitlement)
+        print 'Harry Rein'
+        print available_sessions
+        course_entitlement_available_sessions[str(course_entitlement.uuid)] = available_sessions
+        if not course_entitlement.enrollment_course_run:
+            # Unfulfilled entitlements need a mock session for metadata
+            mock_session = get_mock_session_for_entitlement(course_entitlement)
```

Comment:
Nit: I am not sure mock is a good name here, only because of Mocks use within Python testing.  It might cause confusion.

Reply:
I believe there are other places in the code base that use the same language without directly using mock objects, often when httpretty is being utilized.

Reply:
HttpPretty is another test framework, so I am assuming those are also tests.  This is not test code.

Reply:
Ohhh gotcha. Sorry I should've looked at the context. That's my bad. I agree, mock should probably be changed to something like substitute

Reply:
Changed to pseudo (definition for the english nerds: Pseudo - not actually but having the appearance of; pretended; false or spurious; sham. 2. almost, approaching, or trying to be.)